pkg/inject/envoy_test.go

```diff
@@ -2696,6 +2696,140 @@ func Test_envoyMutator_mutate(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "enabled waitUntilProxyReady",
+			fields: fields{
+				vn: vn,
+				ms: ms,
+				mutatorConfig: envoyMutatorConfig{
+					awsRegion:                  "us-west-2",
+					preview:                    true,
+					logLevel:                   "debug",
+					adminAccessPort:            9901,
+					preStopDelay:               "20",
+					postStartInterval:          5,
+					postStartTimeout:           60,
+					readinessProbeInitialDelay: 10,
+					readinessProbePeriod:       2,
+					sidecarImage:               "envoy:v2",
+					sidecarCPURequests:         cpuRequests.String(),
+					sidecarMemoryRequests:      memoryRequests.String(),
+					waitUntilProxyReady:        true,
+					controllerVersion:          "v1.4.1",
+					k8sVersion:                 "v1.20.1-eks-fdsedv",
+				},
+			},
+			args: args{
+				pod: pod,
+			},
+			wantPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "my-ns",
+					Name:      "my-pod",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "envoy",
+							Image: "envoy:v2",
+							SecurityContext: &corev1.SecurityContext{
+								RunAsUser: aws.Int64(1337),
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "stats",
+									ContainerPort: 9901,
+									Protocol:      "TCP",
+								},
+							},
+							Lifecycle: &corev1.Lifecycle{
+								PostStart: &corev1.Handler{
+									Exec: &corev1.ExecAction{Command: []string{
+										"sh", "-c", "if [ -f /tmp/agent.sock ]; then APPNET_AGENT_POLL_ENVOY_READINESS_TIMEOUT_S=60 " +
+											"APPNET_AGENT_POLL_ENVOY_READINESS_INTERVAL_S=5 /usr/bin/agent -envoyReadiness; fi",
```

Comment:
what are the cases where /tmp/agent.sock might not exist?

Reply:
discussed offline -- we decided to use the envoy version to indicate that postStart hook should only run on version 1.23+